secure-api-gateway-ig-extensions/src/test/java/com/forgerock/sapi/gateway/mtls/DefaultTransportCertValidatorTest.java

```diff
@@ -15,111 +15,172 @@
  */
 package com.forgerock.sapi.gateway.mtls;
 
+import static com.forgerock.sapi.gateway.mtls.DefaultTransportCertValidator.Heaplet.CONFIG_TRANSPORT_KEY_USE;
 import static com.forgerock.sapi.gateway.util.CryptoUtils.generateExpiredX509Cert;
 import static com.forgerock.sapi.gateway.util.CryptoUtils.generateRsaKeyPair;
 import static com.forgerock.sapi.gateway.util.CryptoUtils.generateX509Cert;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.forgerock.json.JsonValue.field;
 import static org.forgerock.json.JsonValue.json;
 import static org.forgerock.json.JsonValue.object;
+import static org.forgerock.secrets.Purpose.purpose;
+import static org.forgerock.secrets.jwkset.JwkSetSecretStore.JwkPredicates.keyUse;
 
 import java.security.cert.CertificateException;
 import java.security.cert.X509Certificate;
 import java.util.Calendar;
 import java.util.Date;
+import java.util.stream.Stream;
 
+import org.forgerock.json.JsonValue;
 import org.forgerock.json.jose.jwk.JWKSet;
 import org.forgerock.openig.heap.HeapImpl;
 import org.forgerock.openig.heap.Name;
+import org.forgerock.secrets.jwkset.JwkSetSecretStore;
+import org.forgerock.secrets.keys.CertificateVerificationKey;
+import org.forgerock.util.Options;
 import org.forgerock.util.Pair;
-import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
 
 import com.forgerock.sapi.gateway.mtls.DefaultTransportCertValidator.Heaplet;
 import com.forgerock.sapi.gateway.util.CryptoUtils;
 
 class DefaultTransportCertValidatorTest {
 
+    // TEST_TLS_CERT actual X509 certificate
```

Comment:
javadoc comments instead of code comments ?
```suggestion
    /** Actual X509 certificate. */
```

Reply:
Ooops, missed that.